File: time-complexity/missing-element.rb
# Given an array for integers from 1 to N with one integer missing, find the missing integer.

def solution(a)

  a.sort!

  i = 0
  while i < a.length do
    break if a[i] == i + 2
    i += 1
  end

  i + 1

end

File: time-complexity/missing-element.py
# Given an array for integers from 1 to N with one integer missing, find the missing integer.

def solution(A):

    n = range(1,len(A)+2)
    availableValues = {}

    for i in n:
        availableValues[i] = True

    for i in A:
        availableValues.pop(i, None)

    for key in availableValues:
        return key

